### DISCUSSION TITLE
Add Coordinate Output to Motion Detector

### DISCUSSION THREAD

**Discussion by gerakis** [FakeFishGames/Barotrauma#7906] [Ideas and suggestions]
Add a few more pins to motion detectors that give the relative X and Y coordinates of the thing that activated them, as well as the degree heading from the motion detector's center as well as the distance. So an X pin, a Y pin, a degrees pin, and a distance pin.

I want to make motion activated guns better. Its possible to make a primitive one right now that can shoot only at right angles. I want one that can shoot at any angle. I think people would love that. If you think its overpowered just don't add it to vanilla subs. Its not much different than just assigning a bot to man the guns.

You could do other things with this too, like create seachlights that track anything moving. That would be a very creepy effect when approaching a wreck or a pirate sub. You swim towards it only to get slammed with a light as soon as you are within range. You could make the creepiest cargo hold ever; one that only illuminates where you are standing with a single beam that sweeps across the hold as you walk. Everything else is pitch black.

Also, a Minerals option would be nice too, (any, humans, monsters, walls, minerals) just to shine a searchlight on any mineral patches when mining. I think people would appreciate that subtle touch.

**Comment by mygamingaccount** [comment]
An array of detectors that transmit their own bearing to the turret would be enough. You could calculate a parallax correction if you wanted to aim multiple turrets.

I suggest that you join #baro-sub-building so we can further elaborate about this, it's a better forum for this kind of discussion.

**Reply by MaaiG** [reply]
> An array of detectors

That’s not enough.
1. Short range
2. Bulky construction

**Reply by gerakis** [reply]
Then you need dozens of components to control the turret, instead of just 1. Plus the ones at diagonals from the turret probably have to be mounted out in the water away from the ship I'm guessing? More pins would solve a lot of problems.

**Reply by mygamingaccount** [reply]
So? You want to do something that the game was never designed for. Get creative! There are easy ways to solve this. Let's discuss this on #baro-sub-building!

**Reply by gerakis** [reply]
Yes. I want them to change the game so that it becomes something the game was designed for. That is the entire point of making suggestions.

Also I joined baro sub building and nobody knew anything about turrets.

**Reply by MaaiG** [reply]
> Get creative! There are easy ways to solve this.

Yea. But I think it’s from despair. 
![Image 001_16-17](https://user-images.githubusercontent.com/93348499/147477568-4e86bbc5-aa29-4151-a987-980e7061b0d7.png)

https://user-images.githubusercontent.com/93348499/147572473-06918f54-69f9-447e-ad38-c1a033a35777.mp4

I don’t want invisible motion detectors hanging around from my submarine...
By the way, I [found a bug](https://github.com/Regalis11/Barotrauma/issues/7941).

Besides, I’m afraid that creativity can’t solve all the problems right now. The sub-editor already limits creativity.

For example: I can’t use information that the room or compartment has been cut off energy! I can find out that the reactor doesn’t generate power, but I can’t use information on how much energy is coming into the compartment.

There are no tools in the editor:
- for group transfer of wires
- to insert components in the middle of an existing wire
- etc.
- Air does not leave an open airlock... ((

> So? You want to do something that the game was never designed for.

I think the game is primarily about making your own submarine!
And it fits into this idea! It includes such simple things. I don’t need ready-made devices I need well-functioning tools!

Anyway, I feel like I’ve exhausted this constructor.
It’s sad that the emphasis is on the "ugly pet" instead of the submarine and the underwater world. 😢 IMHO

**Reply by gerakis** [reply]
My understanding is you can actually hide a scaled down battery with tiny storage and use that as a "power-is-present" detector. Just use its current charge level. If its higher than 0 its got power, otherwise its cut off somehow. Its pretty ghetto but this is what people are doing from what ive gathered. I havent tested this myself.

**Reply by MaaiG** [reply]
> hide a scaled down battery

Duct tape. 😉

**Reply by leonard84** [reply]
Just for reference https://steamcommunity.com/sharedfiles/filedetails/?id=2185206376

**Reply by MateusJK15** [reply]
With the "align to grid" option it's possible to align a motion sensor to a turret perfectly, so if a motion sensor outputs the bearing of the target to a turret, it would be able hit the target perfectly, but wouldn't be OP since at long range this system would be inaccurate because it wouldn't calculate the leading to target.

**Reply by gerakis** [reply]
Motion sensors can only output a true or false. So if you have giant coverage areas it only lets you know that something is in that area, not where it is.

Try to make an autofire gun. You will see what I'm talking about instantly.

**Comment by MaaiG** [comment]
> Add a few more pins to motion detectors

And I’ve done this. )) But I think We need a new component - the "radar component".
You don't agree?

**Reply by gerakis** [reply]
What would it do differently than a motion detector? Program wise im assuming the motion detector just detects a collision between two rectangles. All they would have to do is output the x,y coordinates of that object to the motion detector and subtract the detectors current coordinates.

**Reply by MaaiG** [reply]
Different logic works.
I don’t mind if the motion detector has that capability, but it makes sense to create a new sensor.
I think, the name of the device shall be fully appropriate to its purpose.

**Reply by FalcoGer** [reply]
motion detectors detect motion. radars detect objects, even non moving ones. though radars underwater make little sense, radar penetration in water is terrible. we have sonar already though.

**Comment by Ek-Videogames** [comment]
It would also be nice if the max-range limits were a bit bigger when setting them in sub-editor, so we don't have to do as much cursed daisy-chain wiring job to make bigger arrays.
